HIP/hip-583.md

```diff
@@ -10,220 +10,518 @@ status: Accepted
+The Mirror Node can already ingest account creation transactions with alias values. This will only increase the
+applicable parent transactions that can cause this flow. Where it was limited to CryptoTransfer it will now include
+CryptoCreate, and in both cases it will now support `evm address` formatted aliases.
 
-`CryptoTransfer` auto-account creation is un-impacted and will still support the provision of the public key. 
+The following table describes how the `alias` and `evm address` fields will appear depending on how the alias was given during account creation.
+A guiding principle is that the `evm address` field will be shown as the address as it appears to the evm.
 
-### Mirror Node Impications
+| Given Account Alias                               | Resultant Evm Address                | Derived |
+|---------------------------------------------------|--------------------------------------|---------|
+| `evm address`                                     | `evm address`                        | False   |
+| ECDSA secp256k1 `key alias`                       | derived `evm address` from given key | True    |
+| Any other alias - ED25519, mutlisig, complex etc. | associated `account-num alias`       | True    |
 
-The Mirror Node can already ingest account creation transactions with alias values.
-This will only increate the applicable parent trnasactions that can cause this flow.
-Where it was limited to CryptoTransfer it will now include CryptoCreate
 
 ### Wallet/Exchange Implications
 
-With the suggested changes, the alias property of an account will become more applicable and will need greater visibility.
+With the suggested changes, the alias property of an account will become more applicable and will need greater
+visibility.
 
-To provide greater web3 user scenario support Wallets & Exchanges should
+To provide greater web3 user scenario support Wallets & Exchanges should:
 
 - Display the account alias property of an account
 - Create an ECDSA based account with alias when full smart contract support is desired
-- Accept the use of an alias as a account identifier in HBAR and token transfers
+- Accept the use of an alias as an account identifier in HBAR and token transfers
 - Accept the use of an alias during account creation
 - Add support to check if an account exists with a given alias
 
 ### Performance Implications
 
-The mass adoption of alias setting on new and existing accounts will have minor but notable memory implications on the Hedera ecosystem
+`CryptoCreate` transactions will see extra bytes for every account created with an alias.
 
-`CryptoCreate` transactions will see extra bytes for every account created with an alias
+The size of the bytes in each case depends on the governing account key type. For an `ECDSA` based account the alias
+will always be a valid Ethereum address and will be 20 bytes long. In the `ED25519` case the alias will be a valid
```

Comment:
The `ED25519` alias part can be removed since it will not be supported for `CryptoCreate`.